docs/dev-guide/debugging.rst

```diff
@@ -0,0 +1,46 @@
+Debugging
+=========
+
+.. _runtime_permormance:
+
+Runtime Performance Analysis
+----------------------------
+
+Python standard library provides a `cProfile` module to assist developers in collecting a set of
+statistics that describes how often and for how long various parts of the program executed.
+Full documentation for this module can be found
+`here <https://docs.python.org/2/library/profile.htm>`_. To capture information about all function
+calls simply add the following lines of code right before the code you are interested in
+analyzing::
+
+    import cProfile
+    pr = cProfile.Profile()
+    pr.enable()
+
+The above code snippet is then followed by any code you are analyzing. Then after the code that is
```

Comment:
Consider removing the first sentence since that was stated above.
s/Then after/After/